app/scripts/controllers/transactions/index.js

```diff
@@ -37,6 +37,7 @@ import { hexToBn, bnToHex, BnMultiplyByFraction } from '../../lib/util'
 import { TRANSACTION_NO_CONTRACT_ERROR_KEY } from '../../../../ui/app/helpers/constants/error-keys'
 
 const SIMPLE_GAS_COST = '0x5208' // Hex for 21000, cost of a simple send.
+const MAX_MEMSTORE_TX_LIST_SIZE = 100 // Number of transactions (by unique nonces) to keep in memory
```

Comment:
Is this number too conservative for our power users? I don't have anywhere near 3000 transactions, so I don't know what they might want/need. Presumably our UI can handle 150-200 without slowing down, if 100 took (IIRC) 90ms.

I am working on fixing the state log explorer, so they should have that by 8.0.5, regardless.

Reply:
I don't know, I don't think there's _a lot_ of utility in looking at the tx details 100+ txs back, but yeah I'm just guessing. A user can still go to Etherscan/their favourite block explorer to see details past this.

Reply:
We can also change this in the future since this doesn't change how things are persisted.